src/main.rs

```diff
@@ -189,7 +190,7 @@ fn main() {
 
     let root = matches
         .value_of("root")
-        .map(PathBuf::from)
+        .map(|s| PathBuf::from(s).canonicalize().unwrap())
```

Comment:
Have you tested it on Windows, `.canonicalize()` may have problem on Windows.

Reply:
I'm on Windows. What exactly is it that you want to test?

Reply:
Serve a relative directory `./`, and open a relative directory under it `/foo/../`

Reply:
Tried serving the repo directory using b93123529531a3922bc19316e427d7fa1208feca.

Opening `/src/../` redirects to `/`. `/src/../src/` redirects to `/src`. Seems like valid paths work fine.

Opening `/../src/` redirects to `/src/`. Is it supposed to do that, instead of returning 404?

Reply:
`/../src/` is not permitted to access, whether it exists or not, because it is not a sub-directory of `root`. I think should return 403 in this case.

Reply:
On Linux  `cd ../` in `/` leaves you in root.  Replace `src` with `boot` and try in your root. `/../bin` puts you in `/bin/`. I could change the behavior, but I personally prefer to be consistent with filesystems.

Reply:
> On Linux `cd ../` in `/` leaves you in root.  Replace `src` with `boot` and try in your root. `/../bin` puts you in `/bin/`

Same on Windows

`/../simple-http-server` returns 404, so this doesn't leak information about other directories.